### ISSUE TITLE
Cannot edit cell value in text input as edit template

### ISSUE THREAD

**Issue by wnvko** [IgniteUI/igniteui-react#61]
## Description  
I want to use `IgrInput`, type text, as cell edit template.

## Steps to reproduce  
Add `IgrInput` as cell edit template in the grid. I am using this code:
```tsx
const textInputInlineEditorTemplate = (ctx: { dataContext: IgrCellTemplateContext }) => {
  return (
    <>
      <IgrInput type="text" value={ctx.dataContext.cell.editValue} change={(_s: IgrInput, e: IgrComponentValueChangedEventArgs) => ctx.dataContext.cell.editValue = e.detail}></IgrInput>
    </>
  );
}

return (
  <>
    <div className={classes("column-layout master-view-container")}>
      <IgrGrid data={people} primaryKey="id" rowEditable="true" autoGenerate="false">
        <IgrColumn field="first_name" dataType="string" inlineEditorTemplate={textInputInlineEditorTemplate}></IgrColumn>
      </IgrGrid>
    </div>
  </>
);
```

## Result  
Puttin the cell, with `IgrInput` as cell edit template, in edit mode the input appears with value equal to the cell value. This is ok. The input value is not editable at all.
_Note: if the type of `IgrInput` is `number` it is working as expected!_

## Expected result  
Should be able to use `IgrInput` as cell editor and to update the cell value.

**Comment by MayaKirova** [comment]
Related to https://github.com/IgniteUI/igniteui-angular/issues/14020

**Comment by dkamburov** [comment]
Cherry-picked to blazor branch https://github.com/IgniteUI/igniteui-angular/pull/12125/commits/758c8fc4962725964da53f021320812b1ebfabeb

**Comment by MayaKirova** [comment]
@wnvko Fix is available in internal build 23.2.607 and will be available in the next public release.
